app/webpacker/controllers/product_preview_controller.js

```diff
@@ -2,13 +2,8 @@ import { Controller } from "stimulus";
 
 export default class extends Controller {
   connect() {
-    // open the modal before rendering the html to avoid opening a blank modal
-    // TODO other option is having a controller in the stream html that would dispatch the open element on connect
-    window.addEventListener("turbo:before-stream-render", this.#open.bind(this));
-  }
-
-  disconnect() {
-    window.removeEventListener("turbo:before-stream-render", this.#open.bind(this));
+    // open the modal when html is rendering to avoid opening a blank modal
+    this.#open()
```

Comment:
I was wondering if the blank modal is an indicator of _something happening_. If the request takes some time, how does the user know that the preview is on its way?

So I was wondering if we should open the modal but if modals should always be initialised with the _loading_ spinner. I haven't played around with it thought. Not sure if it makes sense.

Reply:
Good idea, it's important to show a loading state whenever we're waiting for a server response. However in the final solution, we went with loading the whole modal on-demand, which makes that harder.

With the main turbo-frame on this page, we show the loading state by giving the frame a loading spinner as the initial content. Turbo knows to display this whenever we're loading something (when it's in a `busy` state).

Maybe we could add custom JS set this to a busy state.
Or maybe Turbo's default blue loading indicator shows already? Or does the browser loading state show (spinner next to tab title)?

Or maybe we go back to the original solution of having the modal present in the page, and giving it a 'busy' state with loading spinner like Maikel suggests. I think we could make a generic modal loader like that (if we don't already have a generic solution for that..?)

So.. in summary I hope that we already have  simple solution for that. If not, it might be fine to go without a loading indicator (maybe the preview loads really quickly anyway).

Reply:
FYI we have the same issue with the "edit image modal". 

So I had play with this, and the turbo loading bar shows at the top of the screen (blue progress bar) but it's not super obvious... 

> Or does the browser loading state show (spinner next to tab title)?

Turbo is not a typical HTTP request so no spinner from browser.

Having generic modal with a loading spinner as default would be nice, and we are bound to need this down the line, but I think we could leave as it is for now, it should load pretty quickly unless you have tons of variants.
What so you guys think ?

Reply:
I'd be happy with the blue turbo loading bar for now.

But if you have funded time left in the budget, then please go ahead with a generic modal solution!

Reply:
I'm happy with the current solution. I just flagged it as nice to have and because there seemed to be some effort in getting rid of a potential loading indicator.